modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/type/IgniteTypeFactory.java

```diff
@@ -114,22 +116,7 @@ public IgniteTypeFactory(RelDataTypeSystem typeSystem) {
             charset = StandardCharsets.UTF_8;
         }
 
-        // IgniteCustomType: all custom data types are registered here
-        NewCustomType uuidType = new NewCustomType(UuidType.SPEC, (nullable, precision) -> new UuidType(nullable));
-        // UUID type can be converted from character types.
-        uuidType.addCoercionRules(SqlTypeName.CHAR_TYPES);
-
-        customDataTypes = new CustomDataTypes(Set.of(uuidType));
-    }
-
-    /** {@inheritDoc} */
-    @Override
-    public RelDataType createSqlType(SqlTypeName typeName) {
-        if (typeName == SqlTypeName.UUID) {
-            return createCustomType(UuidType.NAME);
-        } else {
-            return super.createSqlType(typeName);
-        }
+        customDataTypes = new CustomDataTypes(Set.of());
```

Comment:
Should we drop CustomDataTypes class?
And org.apache.ignite.internal.sql.engine.util.SafeCustomTypeInternalConversion ?

Reply:
I thought about it and decided and postpone clean up, because we might want to introduce new types in the nearest future. Anyway I filed https://issues.apache.org/jira/browse/IGNITE-25230 to return to this later. If it still won't be used, then we remove this code completely.